hunting/__main__.py

```diff
@@ -0,0 +1,278 @@
+# Copyright Elasticsearch B.V. and/or licensed to Elasticsearch B.V. under one
+# or more contributor license agreements. Licensed under the Elastic License
+# 2.0; you may not use this file except in compliance with the Elastic License
+# 2.0.
+
+import json
+import textwrap
+from dataclasses import asdict
+from pathlib import Path
+
+import click
+from tabulate import tabulate
+
+from detection_rules.misc import parse_user_config
+
+from .definitions import HUNTING_DIR
+from .markdown import MarkdownGenerator
+from .run import QueryRunner
+from .search import QueryIndex
+from .utils import (filter_elasticsearch_params, get_hunt_path, load_all_toml,
+                    load_toml, update_index_yml)
+
+
+@click.group()
+def hunting():
+    """Commands for managing hunting queries and converting TOML to Markdown."""
+    pass
+
+
+@hunting.command('generate-markdown')
+@click.argument('path', required=False)
+def generate_markdown(path: click.Path):
+    """Convert TOML hunting queries to Markdown format."""
+    markdown_generator = MarkdownGenerator(HUNTING_DIR)
+
+    if path:
+        path = Path(path)
+        if path.is_file() and path.suffix == '.toml':
+            click.echo(f"Generating Markdown for single file: {path}")
+            markdown_generator.process_file(path)
+        elif (HUNTING_DIR / path).is_dir():
+            click.echo(f"Generating Markdown for folder: {path}")
+            markdown_generator.process_folder(path)
+        else:
+            click.echo(f"Invalid path: {path}. It should be a valid TOML file or a folder.")
```

Comment:
It would be good to `raise ValueError("....")` here to stop the execution flow -- the path was provided but it is not valid

Reply:
Fixed!